stl/inc/span

```diff
@@ -226,28 +226,28 @@ struct pointer_traits<_Span_iterator<_Ty>> {
 };
 
 // STRUCT TEMPLATE _Span_extent_type
-template <size_t _Extent>
+template <class _Ty, size_t _Extent>
 struct _Span_extent_type {
+    using pointer = _Ty*;
+
     constexpr _Span_extent_type() noexcept = default;
 
-    constexpr explicit _Span_extent_type(size_t) noexcept {}
+    constexpr explicit _Span_extent_type(size_t, const pointer _Data) noexcept : _Mydata{_Data} {}

```

Comment:
I observe that the constructor parameters are now reversing the public parameters and the data members, which seems unnecessarily confusing. (Earlier this order made sense due to the layout.) Ditto the other specialization.